### ISSUE TITLE
Perch2 Simple speedup

### ISSUE THREAD

**Issue by louisfh** [kitzeslab/bioacoustics-model-zoo#49]
There's a big speed-up for Perch2 (and maybe other models?) that can be implemented pretty simply.

<h2>Summary</h2>
<p>In <code>Perch2.__call__</code> per-batch outputs are converted to Python lists via <code>.tolist()</code>, accumulated, then re-converted to numpy via <code>np.array(list)</code> at the end. With preds for all ~15,000 perch2 classes this is ~40% of total inference wall time. Also it silently upcasts the model's native float32 to float64.</p>

I used py-spy to profile my big run of perch2 preds, and see why it was taking so slow
`py-spy --native 120` on snowy  shows 

% wall time | Frame
-- | --
26% | np.array(list) at end of __call__ (PyArray_FromAny → DOUBLE_setitem)
17% | .tolist() per batch (PyArray_ToList → recursive_tolist)
12% | actual TF model forward pass

<h2>Fix</h2>
<p>In <code>__call__</code>, accumulate numpy arrays per batch and <code>np.concatenate</code> once at the end instead of going through Python lists:</p>
<pre><code class="language-python"># before
returns["logits"].extend(outs["label"].tolist())
...
return np.array(returns["logits"])

# after
returns["logits"].append(outs["label"])
...
return np.concatenate(returns["logits"], axis=0)
</code></pre>
<p>Same change applies to <code>embeddings</code>, <code>spatial_embeddings</code>, <code>spectrograms</code>, and <code>custom_classifier_logits</code>.</p>
<h2>Measured impact</h2>
<p>I ran 100 WAVs from a real ARU dataset on snowy. batch_size=16, num_workers=8:</p>
<ul>
<li><strong>Speed:</strong> 75.5s → 45.8s (<strong>1.65× speedup</strong>)</li>
<li><strong>Memory:</strong> outputs are now float32 (model's native dtype) instead of float64 — halves output memory</li>
<li><strong>Correctness:</strong> logits agree to float32 precision (max abs diff 4.5e-3 on values up to ±15). </li>
</ul>
<p>I can open a PR to fix, as I've patched it locally.</p></body></html>

**Comment by sammlapp** [comment]
thanks, please do open the PR @louisfh
